build-and-test.ps1

```diff
@@ -0,0 +1,69 @@
+[cmdletbinding()]
+param(
+    [string]$VersionFilter = "*",
+    [string]$OSFilter = "*",
+    [switch]$IsDryRun
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+$buildFilter = "$VersionFilter-$OSFilter/*"
+
+function Get-ImageBuilder()
+{
+    $imageBuilderImageName = 'microsoft/dotnet-buildtools-prereqs:image-builder-nanoserver-20180105103709'
+    $imageBuilderContainerName = "ImageBuilder-$(Get-Date -Format yyyyMMddhhmmss)"
+    New-Item -Path "$imageBuilderFolder" -ItemType Directory -Force | Out-Null
+
+    $attempt = 0
```

Comment:
I think it would help the readability to factor out the logic to perform the retry into a separate function.